merge-intervals/delight010.swift

```diff
@@ -0,0 +1,31 @@
+class Solution {
+    // Time O(n log n)
+    // Space O(n)
+    func merge(_ intervals: [[Int]]) -> [[Int]] {
+        if intervals.count <= 1 {
+            return intervals
+        }
+        
+        let intervals = intervals.sorted { $0[0] <= $1[0] }
+        var answer: [[Int]] = []
+        var currentInterval: [Int] = []
+        
+        for i in 0..<intervals.count {
+            if currentInterval.isEmpty {
+                currentInterval = intervals[i]
+            }
+            
+            if intervals[i][0] >= currentInterval[0] && intervals[i][0] <= currentInterval[1] {
+                currentInterval = [currentInterval[0], max(currentInterval[1], intervals[i][1])]
```

Comment:
```suggestion
            if intervals[i][0] <= currentInterval[1] {
                currentInterval[1] = intervals[i][1]
```
`[0]` 기준으로 정렬한 상태에서 `intervals[i][0] >= currentInterval[0]`은 항상 참일 것 같아요.
왜냐하면 `currentInterval = intervals[j]`라고 하면 `j <= i`이므로 `intervals[i][0] >= intervals[j][0]`가 참이기 때문이에요.
또한 `max` 연산도 생략 가능해 보입니다.